javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -232,4 +249,24 @@ private static String getSerializedName(ProxyMethodParameter parameter) {
         }
         return serializedName;
     }
+
+    private List<ProxyMethodParameter> getProxyMethodParameters(
+            ProxyMethod proxyMethod,
+            List<ClientMethodParameter> clientMethodParameters) {
+        // the list of proxy method parameters will be 1-1 with list of client method parameters
+
+        Map<String, ProxyMethodParameter> proxyMethodParameterByClientParameterName = proxyMethod.getParameters().stream()
+                .collect(Collectors.toMap(p -> CodeNamer.getEscapedReservedClientMethodParameterName(p.getName()), Function.identity()));
+        List<ProxyMethodParameter> proxyMethodParameters = new ArrayList<>();
+        for (ClientMethodParameter clientMethodParameter : clientMethodParameters) {
+            ProxyMethodParameter proxyMethodParameter = proxyMethodParameterByClientParameterName.get(clientMethodParameter.getName());
+            proxyMethodParameters.add(proxyMethodParameter);
```

Comment:
Maybe we should put `requestParameterLocation` to client method parameter, instead of making the mapping.